### PR TITLE
Add dump build time functionality to Test command

### LINKED ISSUE #33916: Improve bal test command to support --dump-build-time option
**Description:**
Without that we can not get the build time with tests.

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Port dump build time from build to test command

Fixes #33916

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by Dilhasha — DISMISSED** [review]
LGTM

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java [hunk 1]
```diff
@@ -133,10 +144,14 @@ public TestCommand(Path projectPath, PrintStream outStream, PrintStream errStrea
             description = "hidden option for code coverage to include all classes")
     private String includes;
 
+    @CommandLine.Option(names = "--dump-build-time", description = "calculate and dump build time")
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java:147]: This should be a hiddne flag

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help [hunk 2]
```diff
@@ -100,6 +100,9 @@ OPTIONS
            Include the dependencies and artifacts in the  executable JAR file(s) of a 
            Ballerina program that are required to enable the observability functionalities on it.
 
+       --dump-build-time
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help:103]: We shouldn't add descriptions for hidden flags. Related to https://github.com/ballerina-platform/ballerina-lang/pull/33932/files#r754899702

Comment (Ibaqu) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help]: Removed `--dump-build-time` from help.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java (hunk 1 above)
Comment (Ibaqu) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java]: Changed to hidden

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java [hunk 3]
```diff
@@ -156,10 +171,18 @@ public void execute() {
             testReport = false;
         }
         BuildOptions buildOptions = constructBuildOptions();
+
         boolean isSingleFile = false;
         if (FileUtils.hasExtension(this.projectPath)) {
             try {
+                if (buildOptions.dumpBuildTime()) {
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java:178]: Let's take this out of the if condition and avoid repeating the code in both blocks

File: cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help [hunk 4]
```diff
@@ -4,7 +4,7 @@ NAME
 SYNOPSIS
        bal test
        bal test [--debug] <port>
-       bal test [--offline] [--rerun-failed] [--experimental] [--observability-included]
+       bal test [--offline] [--rerun-failed] [--dump-build-time] [--experimental] [--observability-included]
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help:7]: This is a hidden flag. So we dont need to add this here. Please check and update the build command help as well if it has this

Comment (Ibaqu) [reply cli/ballerina-cli/src/main/resources/cli-help/ballerina-test.help]: Removed this from all other command help files

**Review by azinneera — APPROVED** [review]

**Pull request merged**